RulesEngine/RuleProcessor.cs

```diff
@@ -310,45 +310,36 @@ private IEnumerable<Rule> GetRulesForLanguages(string[] languages)
         }
 
         /// <summary>
-        /// Check if rule tags have already been seen or if exception exists
+        /// Check if rule has at least one unique tag not seen before or if exception exists
         /// Assumes that _uniqueTagsOnly == true has been checked first for relevance
         /// </summary>
         /// <param name="ruleTags"></param>
         /// <returns></returns>
         private bool UniqueTagsCheck(string[] ruleTags)
         {
-            bool approved = true;
+            bool approved = false;
 
             foreach (string tag in ruleTags)
             {
-                if (_uniqueTagHashes.Contains(tag))
+                if (!_uniqueTagHashes.Contains(tag))
                 {
-                    approved = false;
-                    if (UniqueTagExceptions != null)
+                    approved = true;
+                    break;
+                }
+                else if (UniqueTagExceptions != null)
+                {
+                    foreach (string tagException in UniqueTagExceptions)
                     {
-                        foreach (string tagException in UniqueTagExceptions)
+                        if (tag.Contains(tagException))
                         {
-                            approved = tag.Contains(tagException);
-                            if (approved)
-                            {
-                                break;
-                            }
+                            approved = true;
+                            break;
                         }
-                    }
-
-                    if (_logger != null && !approved)
-                    {
-                        _logger.Debug(string.Format("Duplicate tag {0} not added", tag));
-                    }
-
-                    break;
+                    }                   
                 }
-                else
+                else if (_logger != null && !approved)
```

Comment:
You don't know if its a duplicate until you've checked everything and determined if its approved or not. Which might occur inside of one of if branches this was attached to.

```suggestion
                
                if (_logger != null && !approved)
```